application_test.go

```diff
@@ -427,3 +428,21 @@ func TestApplicationConfiguration(t *testing.T) {
 	assert.True(t, ok)
 	assert.Equal(t, ErrCodeNotFound, apiErr.ErrCode)
 }
+
+func TestWaitOnApplication(t *testing.T) {
+	endpoint := newFakeMarathonEndpoint(t, nil)
+	defer endpoint.Close()
+
+	err := endpoint.Client.WaitOnApplication(fakeAppName, 0*time.Millisecond)
+	if assert.Error(t, err) {
```

Comment:
I think you should be able to skip this one and use `assert.EqualError` directly.